app/frontend/src/features/profile/edit/EditHostingPreference.tsx

```diff
@@ -249,28 +248,55 @@ export default function HostingPreferenceForm() {
           />
           <Controller
             control={control}
-            defaultValue={user.smokingAllowed}
+            defaultValue={
+              user.smokingAllowed || SmokingLocation.SMOKING_LOCATION_UNKNOWN
+            }
             name="smokingAllowed"
-            render={({ onChange }) => (
+            render={({ onChange, value }) => (
               <Select
-                defaultValue={user.smokingAllowed || SmokingLocation.SMOKING_LOCATION_UNKNOWN}
                 onChange={onChange}
                 label={ACCEPT_SMOKING}
-                name="smokingAllowed"
                 className={classes.field}
-              >
-                {[
-                  SmokingLocation.SMOKING_LOCATION_UNKNOWN,
-                  SmokingLocation.SMOKING_LOCATION_YES,
-                  SmokingLocation.SMOKING_LOCATION_WINDOW,
-                  SmokingLocation.SMOKING_LOCATION_OUTSIDE,
-                  SmokingLocation.SMOKING_LOCATION_NO,
-                ].map((o) => (
-                  <option value={o} key={o}>
-                    {smokingLocationLabels[o]}
-                  </option>
-                ))}
-              </Select>
+                value={value}
+                id="smokingAllowed"
+                options={[
+                  {
+                    value: SmokingLocation.SMOKING_LOCATION_UNKNOWN,
+                    label:
+                      smokingLocationLabels[
+                        SmokingLocation.SMOKING_LOCATION_UNKNOWN
+                      ],
+                  },
+                  {
+                    value: SmokingLocation.SMOKING_LOCATION_YES,
+                    label:
+                      smokingLocationLabels[
+                        SmokingLocation.SMOKING_LOCATION_YES
+                      ],
+                  },
+                  {
+                    value: SmokingLocation.SMOKING_LOCATION_WINDOW,
+                    label:
+                      smokingLocationLabels[
+                        SmokingLocation.SMOKING_LOCATION_WINDOW
+                      ],
+                  },
+                  {
+                    value: SmokingLocation.SMOKING_LOCATION_OUTSIDE,
+                    label:
+                      smokingLocationLabels[
+                        SmokingLocation.SMOKING_LOCATION_OUTSIDE
+                      ],
+                  },
+                  {
+                    value: SmokingLocation.SMOKING_LOCATION_NO,
+                    label:
+                      smokingLocationLabels[
+                        SmokingLocation.SMOKING_LOCATION_NO
+                      ],
+                  },
```

Comment:
Haha this is worse...
What if you do `Object.entries(smokingLocationLabels).map(([key, value]) => ({label: value, value: +key}))`?
However that may be  less type safe and one could argue it's more opaque.

You decide if you want to leave it like you have done now, how it was before, or that one-liner.

Reply:
I agree it looks worse, but it is more 'solid' because at least you don't have to write html tags (but still can using the children prop of select). I've tried it with object.entries but then we are less flexibile in the sort order of options and you can't have entries in the labels which you don't want to display in the dropdown. 
Eg. the labels contains both a label for unknown and unspecified while now I only show unknown in the dropdown.

Another option is to pass the select component an array of values and a dictionary (or a getLabel function) from which to retrieve the label for each value.

Reply:
I think that's a good idea - our usual use case is the protobuf enums, and in other cases, it probably won't be much more verbose than any other way